--- urls.py.old 2011-12-10 10:06:09.974681923 +0000
+++ urls.py     2011-12-10 09:59:32.374681923 +0000
@@ -19,7 +19,7 @@
 admin.autodiscover()
 
 urlpatterns = patterns('',
-  ('^admin/(.*)', admin.site.root),
+  ('^admin/(.*)', admin.site.root_path),
   ('^render/?', include('graphite.render.urls')),
   ('^cli/?', include('graphite.cli.urls')),
   ('^composer/?', include('graphite.composer.urls')),
